LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -9,18 +9,16 @@ function logDebug(callback)
 
 function pause(duration) {
     return new Promise(resolve => {
-        setTimeout(() => {
-            resolve();
-        }, duration);
+        setTimeout(resolve, duration);
     });
 }
 
-function dispatchMouseActions(actions, pointerType)
+async function dispatchMouseActions(actions, pointerType)
 {
     if (!window.eventSender)
-        return Promise.reject(new Error("window.eventSender is undefined."));
+        throw new Error("window.eventSender is undefined.");
 
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
         setTimeout(async () => {
             eventSender.dragMode = false;
 
```

Comment:
Can this be flattened further to make more full use of async/await?

e.g. `await new Promise(r => setTimeout(r));`